app/src/androidTest/java/com/github/ravifrancesco/softwaredevproject/GPSTrackerTest.java

```diff
@@ -0,0 +1,53 @@
+package com.github.ravifrancesco.softwaredevproject;
+
+
+import android.content.Context;
+import android.content.Intent;
+
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.MediumTest;
+import androidx.test.rule.ServiceTestRule;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.TimeoutException;
+
+import static org.junit.Assert.assertEquals;
+
+
+@RunWith(AndroidJUnit4.class)
+@MediumTest
+public class GPSTrackerTest {
+
+    @Rule
+    public final ServiceTestRule serviceRule = new ServiceTestRule();
+
+    @Test
```

Comment:
Add some quick comments on what is exactly tested.

Reply:
Done